src/main/java/roboy/dialog/states/ordinaryStates/IntroductionState.java

```diff
@@ -0,0 +1,156 @@
+            // alternatively: Output.useFallback() or Output.sayNothing()
+        }
+
+
+        // 2. get interlocutor object from context
+        // this also should query memory and do other magic
+        Interlocutor person = Context.getInstance().ACTIVE_INTERLOCUTOR.getValue();
+        person.addName(name);
+
+
+        // 3. update interlocutor in context
+        updateInterlocutorInContext(person);
+
+        // 4. check if person is known/familiar
+        if (person.FAMILIAR) {
+
+            // 4a. person known/familiar
+            String retrievedResult = "";
+            RandomList<MemoryNodeModel> nodes = retrieveNodesFromMemoryByIds(person.getRelationships(Neo4jRelationships.FRIEND_OF));
+            if (!nodes.isEmpty()) {
+                retrievedResult = " You are friends with " + nodes.getRandomElement().getProperties().get("name").toString();
+            }
+
+            RelationshipAvailability availability = person.checkRelationshipAvailability(predicates);
+            if (availability == SOME_AVAILABLE) {
+                nextState = (Math.random() < 0.3) ? getTransition(UPDATE_KNOWN_PERSON) : getTransition(LEARN_ABOUT_PERSON);
+            } else if (availability == NONE_AVAILABLE) {
+                nextState = getTransition(LEARN_ABOUT_PERSON);
+            } else {
+                nextState = getTransition(UPDATE_KNOWN_PERSON);
+            }
+
+            // TODO: get some friends or hobbies of the person to make answer more interesting
+            return Output.say("Hey, I know you, " + person.getName() + retrievedResult);
+
+        } else {
+            // 4b. person is not known
+            nextState = getTransition(LEARN_ABOUT_PERSON);
+
+            // TODO: what would you say to a new person?
+            return Output.say(String.format("Nice to meet you, %s!", name)).setSegue(new Segue(Segue.SegueType.DISTRACT, 0.6));
+        }
+    }
+
+    @Override
+    public State getNextState() {
+        return nextState;
+    }
+
+    private String getNameFromInput(Interpretation input) {
+        if (input.getSentenceType().compareTo(Linguistics.SENTENCE_TYPE.STATEMENT) == 0) {
+            String[] tokens = (String[]) input.getFeatures().get(Linguistics.TOKENS);
+            if (tokens.length == 1) {
+                return tokens[0].replace("[", "").replace("]","").toLowerCase();
+            } else {
+                if (input.getFeatures().get(Linguistics.PARSER_RESULT).toString().equals("SUCCESS")) {
+                    List<Triple> result = (List<Triple>) input.getFeatures().get(Linguistics.SEM_TRIPLE);
+                    if (result.size() != 0) {
+                        return result.get(0).object.toLowerCase();
+                    } else {
+                        if (input.getFeatures().get(Linguistics.OBJ_ANSWER) != null) {
+                            String name = input.getFeatures().get(Linguistics.OBJ_ANSWER).toString().toLowerCase();
```

Comment:
duplicate code
please move ``if (input.getFeatures().get(Linguistics.OBJ_ANSWER) != null`` to the end of the function.

Reply:
Done